### PR TITLE
Handle NaN results from Presto server

### PR DESCRIPTION
**Pull request by nezihyigitbasi**
with comments addressed (throw an exception when we see a non-NaN double in the result while we were expecting a string)

### CODE REVIEW

File: driver/prestoresults.d [hunk 1]
```diff
@@ -26,10 +26,19 @@ import presto.client.queryresults : ColumnMetadata;
...
 
-void addToPrestoResultRow(JSONValue columnData, PrestoResultRow result) {
+void addToPrestoResultRow(JSONValue columnData, PrestoResultRow result, string resultDataType) {
     final switch (columnData.type) {
         case JSON_TYPE.STRING:
-            result.addNextValue(Variant(columnData.str));
+            if (resultDataType == "string") {
+                result.addNextValue(Variant(columnData.str));
+            } else {
+                //for some scalar functions that return double, we may get a "NaN" back
```
Comment (markisaa) [inline driver/prestoresults.d]: Nit: I think this is pretty self explanatory. One also need not use a function for NaN to be a valid value.

I'm of the philosophy that in-line comments are most often a sign that more names should be introduced into the code itself. (though that applies to other cases, it just makes me weary of in-line comments in general)

File: driver/prestoresults.d [hunk 2]
```diff
@@ -26,10 +26,19 @@ import presto.client.queryresults : ColumnMetadata;
...
+                if (columnData.str == "NaN") { 
+                    result.addNextValue(Variant(double.nan));
+                } else {
+                    dllEnforce(false, "A non-Nan double is not expected in the result");
```
Comment (markisaa) [inline driver/prestoresults.d:36]: Nit:
If(not equal to NaN) {
DllEnforce
} 
Result.doStuff

Prefer flat code style

**Comment by markisaa** [comment]
Looks great :)

**Comment by andralex** [comment]
Looking good, merging. Thanks!

**Comment by andralex** [comment]
I merged using the cmd line but this doesn't come as closed. This has been an intermittent issue. @electrum seems like something's wrong with our git scripts? Will close manually.

**Pull request closed without merging**